package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -316,6 +316,18 @@ def __init__(self, universe,
 
         self.u = universe
         self._trajectory = self.u.trajectory
+
+        msg = ("{}_sel is an empty selection string - no hydrogen bonds will "
+               "be found. This may be intended, but please check your "
+               "selection."
+               )
+        for sel, sel_string in zip(
```

Comment:
So the line optimising person in me thinks that this would be a fun way to squash a few more lines:

```python
for sel in ['donors_sel', 'hydrogens_sel', 'acceptors_sel']:
    var = eval(sel)
    setattr(self, sel, var)
    if isinstance(var, str) and not var.strip():
        warnings.warn(msg.format(sel))
```

Whilst I _think_ it's readable to me, maybe others would disagree - @lilyminium thoughts?

Reply:
If you didn't want to mess with `eval` you could achieve a similar effect by examining `self`. e.g. (below setting self.donors_sel etc):

```
for sel in ['donors_sel', 'hydrogens_sel', 'acceptors_sel']:
    val = getattr(self, sel)
    if isinstance(val, str) and not val:
        # do the warning
```

Reply:
Should this only be checking for the empty string? Or should you check for all cases where nothing is selected (e.g. `name NOT_A_REAL_ATOM`?) Otherwise it seems to me this PR could work just by changing the check to `if self.xx_sel is None: guess` as you do below, without the warning, and let the selection parser handle the empty string however it does.

Reply:
I think it should only check for an empty string, because some selections might have empty atom groups at some frames but not others (e.g. those that include `around`). Also, relying on the selection parser would give the following response:

`"Empty string to select atoms, empty group returned."`

whereas it might be useful for the user to know which selection string was empty.